src/flyte/_internal/imagebuild/remote_builder.py

```diff
@@ -261,7 +261,10 @@ def _get_layers_proto(image: Image, context_path: Path) -> "image_definition_pb2
                 pyproject_dst = copy_files_to_context(layer.pyproject, context_path)
             else:
                 # Copy the entire project
-                pyproject_dst = copy_files_to_context(layer.pyproject.parent, context_path)
+                docker_ignore_patterns = get_and_list_dockerignore(image)
+                pyproject_dst = copy_files_to_context(
+                    layer.pyproject.parent, context_path, docker_ignore_patterns, ["pyproject.toml", "uv.lock"]
+                )
```

Comment:
I'm wondering if we could just remove "pyproject.toml", "uv.lock" from `docker_ignore_patterns`, so we don't need to add protected_patterns?